medmodels/medrecord/querying.pyi

```diff
@@ -19,314 +20,377 @@ else:
+NodeIndicesComparisonOperand: TypeAlias = Union[NodeIndicesOperand, List[NodeIndex]]
+
 EdgeIndexComparisonOperand: TypeAlias = Union[
     EdgeIndexOperand,
     EdgeIndex,
+]
+EdgeIndicesComparisonOperand: TypeAlias = Union[
     EdgeIndicesOperand,
     List[EdgeIndex],
 ]
-EdgeIndexArithmeticOperand: TypeAlias = Union[EdgeIndexOperand, EdgeIndex]
 
 class EdgeDirection(Enum):
     INCOMING = auto()
     OUTGOING = auto()
     BOTH = auto()
 
 class NodeOperand:
-    def attribute(self, attribute: MedRecordAttribute) -> MultipleValuesOperand: ...
+    def attribute(
+        self, attribute: Union[MedRecordAttribute, SingleAttributeOperand]
+    ) -> MultipleValuesOperand: ...
     def attributes(self) -> MultipleAttributesOperand: ...
     def index(self) -> NodeIndexOperand: ...
     def in_group(self, group: Union[Group, List[Group]]) -> None: ...
     def has_attribute(
-        self, attribute: Union[MedRecordAttribute, List[MedRecordAttribute]]
+        self, attribute: Union[MedRecordAttribute, SingleAttributeOperand]
     ) -> None: ...
     def incoming_edges(self) -> EdgeOperand: ...
     def outgoing_edges(self) -> EdgeOperand: ...
     def neighbors(
         self, edge_direction: EdgeDirection = EdgeDirection.OUTGOING
     ) -> NodeOperand: ...
     def either_or(self, either: NodeQuery, or_: NodeQuery) -> None: ...
+    def clone(self) -> NodeOperand: ...
 
 class EdgeOperand:
-    def attribute(self, attribute: MedRecordAttribute) -> MultipleValuesOperand: ...
+    def attribute(
+        self, attribute: Union[MedRecordAttribute, SingleAttributeOperand]
+    ) -> MultipleValuesOperand: ...
     def attributes(self) -> MultipleAttributesOperand: ...
     def index(self) -> EdgeIndexOperand: ...
     def in_group(self, group: Union[Group, List[Group]]) -> None: ...
     def has_attribute(
-        self, attribute: Union[MedRecordAttribute, List[MedRecordAttribute]]
+        self, attribute: Union[MedRecordAttribute, SingleAttributeOperand]
     ) -> None: ...
     def source_node(self) -> NodeOperand: ...
     def target_node(self) -> NodeOperand: ...
     def either_or(self, either: EdgeQuery, or_: EdgeQuery) -> None: ...
+    def clone(self) -> EdgeOperand: ...
 
 class MultipleValuesOperand:
     def max(self) -> SingleValueOperand: ...
     def min(self) -> SingleValueOperand: ...
+    def first_where(
+        self, query: Callable[[SingleValueOperand], None]
+    ) -> SingleValueOperand: ...
+    def last_where(
+        self, query: Callable[[SingleValueOperand], None]
+    ) -> SingleValueOperand: ...
     def mean(self) -> SingleValueOperand: ...
-    def any(self, query: Callable[[SingleValueOperand], None]) -> None: ...
+    def median(self) -> SingleValueOperand: ...
+    def mode(self) -> SingleValueOperand: ...
+    def std(self) -> SingleValueOperand: ...
+    def var(self) -> SingleValueOperand: ...
+    def count(self) -> SingleValueOperand: ...
```

Comment:
Is the idea behind `count` to show the number of available values?

Reply:
not to show the number of available values (e.g. print or anything) but to use the number of available values in another part of the query, yes